pkg/kopia/command/kopia.go

```diff
@@ -59,3 +71,188 @@ func commonArgs(password, configFilePath, logDirectory string, requireInfoLevel
 func ExecKopiaArgs(configFilePath string) []string {
 	return commonArgs("", configFilePath, "", false).StringSliceCMD()
 }
+
+func ResolveS3Endpoint(endpoint string) string {
+	if strings.HasSuffix(endpoint, "/") {
+		log.Debug().Print(fmt.Sprintf("Removing trailing slashes from the endpoint %s", endpoint))
+		endpoint = strings.TrimRight(endpoint, "/")
+	}
+	sp := strings.SplitN(endpoint, "://", 2)
+	if len(sp) > 1 {
+		log.Debug().Print(fmt.Sprintf("Removing leading protocol from the endpoint %s", endpoint))
+	}
+	return sp[len(sp)-1]
+}
+
+func HttpInsecureEndpoint(endpoint string) bool {
+	return strings.HasPrefix(endpoint, "http:")
+}
+
+// GenerateFullRepoPath defines the manner in which a location-specific prefix
+// string is joined with a repository-specific prefix to generate the full path
+// for a kopia repository.
+func GenerateFullRepoPath(locPrefix string, artifactPrefix string) string {
```

Comment:
```suggestion
func GenerateFullRepoPath(locPrefix string, repoPathPrefix string) string {
```
We should use `repoPathPrefix` instead of `artifactPrefix` going forward. Please rename it here and everywhere else.
@ankitjain235 what do you think?